core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -199,6 +199,12 @@
    @GuardedBy("transferMapsLock")
    protected final Map<Integer, List<InboundTransferTask>> transfersBySegment = new HashMap<>();
 
+   /**
+    * A set identifying the transactional segments requested by the cache. This is a set so a segment is counted only
+    * once.
+    */
+   private Set<Integer> requestedTransactionalSegments;
```

Comment:
We will want to make this an IntSet as well so we can invoke the primitive set method.